lib/ukfallocbuddy/tests/test_ukfallocbuddy.c

```diff
@@ -0,0 +1,217 @@
+#include <uk/test.h>
+#include <uk/fallocbuddy.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+
+struct test_ukfallocbuddy {
+
+	struct uk_falloc *fa;
+	__paddr_t start;
+	__sz len;
+	unsigned long pages;
+	__sz fa_meta_size;
+	void *start_zone;
+	void *fa_meta;
+};
+
+
+static struct uk_falloc *init_new_fb_alloc(void *fa_meta, void *start_zone,
+	unsigned long len, __paddr_t *new_start, __sz fa_meta_size,
+	unsigned long *pages)
+{
+	__vaddr_t dm_off;
+	__paddr_t start = (__paddr_t)start_zone;
+	struct uk_falloc *fa;
+
+	UK_ASSERT(fa = malloc(uk_fallocbuddy_size()));
+
+	UK_ASSERT(!uk_fallocbuddy_init(fa));
+	start = PAGE_ALIGN_UP(start + fa_meta_size);
+	*pages = (len - fa_meta_size) >> PAGE_SHIFT;
+	dm_off = start;
+	UK_ASSERT(!fa->addmem(fa, fa_meta, start, *pages, dm_off));
+	*new_start = start;
+	return fa;
```

Comment:
You cannot use `UK_ASSERT()` to perform any action, because the expression is not evaluated when asserts are disabled. I suggest you add the following helper macro:
```C
#define fab_bug_on(cond)						\
	do {								\
		if (unlikely(cond))					\
			UK_CRASH("'%s' during test execution.\n",	\
				 STRINGIFY(cond));			\
	} while (0)
```
And change the code to this:
```suggestion
	__paddr_t start = (__paddr_t)start_zone;
	struct uk_falloc *fa;
	int rc;

	fa = malloc(uk_fallocbuddy_size());
	fab_bug_on(fa == NULL);

	rc = uk_fallocbuddy_init(fa);
	fab_bug_on(rc != 0);

	start = PAGE_ALIGN_UP(start + fa_meta_size);
	*pages = (len - fa_meta_size) >> PAGE_SHIFT;

	rc = fa->addmem(fa, fa_meta, start, *pages, start);
	fab_bug_on(rc != 0);

	*new_start = start;
	return fa;
```